test/unit/UserManagementApi.spec.ts

```diff
@@ -0,0 +1,141 @@
+
+import * as chai from 'chai';
+import * as sinon from 'sinon';
+import { UserManagementApi } from '../../source';
+import { ApiCredentials } from '@adastradev/serverless-discovery-sdk';
+
+const expect = chai.expect;
+
+describe('UserManagementApi', () => {
+
+    let sandbox: sinon.SinonSandbox;
+    let api: UserManagementApi;
+    const tenant_id: string = 'aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee';
+    const id: string = 'idididid';
+    const tenantName: string = 'blah';
+    const userName: string = 'testusername';
+    const password: string = 'testpassword';
+    const firstName: string = 'blah';
+    const lastName: string = 'blah';
+    let client: {
+        invokeApi: sinon.SinonStub
+    };
+
+    beforeEach(() => {
+        sandbox = sinon.createSandbox();
+        client = { invokeApi: sinon.stub() };
+        api = new UserManagementApi('blah', 'blah', { type: 'None' } as ApiCredentials, client);
+    });
+
+    afterEach(() => {
+        sandbox.restore();
+    });
+
+    describe('createUserPool', () => {
+        it('Should invoke api with correct args', async () => {
+            await api.createUserPool(tenant_id);
+            expect(client.invokeApi.calledOnce).to.be.true;
+            expect(client.invokeApi.args[0][0]).to.deep.equal({});
+            expect(client.invokeApi.args[0][1]).to.deep.equal('/admin/userpools');
+            expect(client.invokeApi.args[0][2]).to.deep.equal('POST');
+            expect(client.invokeApi.args[0][3]).to.deep.equal({});
+            expect(client.invokeApi.args[0][4]).to.deep.equal({ tenant_id });
+        });
+    });
+
+    describe('deleteUserPool', () => {
+        it('Should invoke api with correct args', async () => {
+            await api.deleteUserPool(id);
+            expect(client.invokeApi.calledOnce).to.be.true;
+            expect(client.invokeApi.args[0][0]).to.deep.equal({});
```

Comment:
You might be able to collapse these lines into one .deep.equal by comparing args[0] to a whole object you build prior to the expect